feat!: migrate to direct agentic API (v2) (#334)
markupai/content-guardian-action: README.md

[![Coverage](https://github.com/markupai/content-guardian-action/blob/main/badges/coverage.svg)](https://github.com/markupai/content-guardian-action)

A GitHub Action that analyzes commit changes and runs style checks on modified
files via the Markup AI style agent. Automatically adapts to push, pull request,

> manual, and scheduled events.

Author: **Markup AI** detected issues:
- **Consistency / Is your comma usage consistent? (Severity: High)**: `,`
*Explanation: Could you remove the comma before &#34;and&#34; or &#34;or&#34;? Remove the comma before the conjunction unless one of the elements of the series contains a conjunction.*
```suggestion
manual and and scheduled events.
```

> ## What's new in v2

Author: **Markup AI** detected issues:
- **Consistency / Is your title case consistent? (Severity: High)**: `new`
*Explanation: Could you use the AP Stylebook for title case? Capitalize most words in titles.*
```suggestion
## What's New in v2
```

- **Direct agentic API** — no longer uses `@markupai/toolkit`; talks to
  `https://api.markup.ai` directly.
- **Single `target` input** — replaces v1's `dialect` / `tone` / `style-guide`
  combo. Pass a target ID or its display name (look up at
  [console.markup.ai](https://console.markup.ai)).
- **Risk-based scoring by default** — if your organization has numeric scoring
  disabled, the action surfaces severity counts and a risk label instead of
  fabricated scores. Numeric scoring is shown only when your org enables it.

> Migrating from v1: drop `dialect`, `tone`, and `style-guide`; add `target`.

Author: **Markup AI** detected issues:
- **Consistency / Is your comma usage consistent? (Severity: High)**: `,`
*Explanation: Could you remove the comma before &#34;and&#34; or &#34;or&#34;? Remove the comma before the conjunction unless one of the elements of the series contains a conjunction.*
```suggestion
Migrating from v1: drop `dialect`, `tone` and and `style-guide`; add `target`.
```

## Features

- 🔍 **Smart File Discovery**: Detects files to analyze based on the GitHub event
- 📝 **Event-Based Analysis**: Optimized behavior for push, pull request,

>   manual, and scheduled events

Author: **Markup AI** detected issues:
- **Consistency / Is your comma usage consistent? (Severity: High)**: `,`
*Explanation: Could you remove the comma before &#34;and&#34; or &#34;or&#34;? Remove the comma before the conjunction unless one of the elements of the series contains a conjunction.*
```suggestion
  manual and and scheduled events
```

- 📊 **Risk or numeric scoring**: Severity counts (high/medium/low) plus an
  overall risk label, or a 0–100 quality score, driven by your org config
- 🏷️ **Visual feedback**: Commit status updates, PR comments, inline review
  suggestions
- 📋 **Rich outputs**: JSON results and detailed reporting

> ## Supported file types

Author: **Markup AI** detected issues:
- **Consistency / Is your title case consistent? (Severity: High)**: `file`
*Explanation: Could you use the AP Stylebook for title case? Capitalize most words in titles.*
```suggestion
## Supported File types
```
- **Consistency / Is your title case consistent? (Severity: High)**: `types`
*Explanation: Could you use the AP Stylebook for title case? Capitalize most words in titles.*
```suggestion
## Supported file Types
```

## Usage

> ### Basic usage

Author: **Markup AI** detected issues:
- **Consistency / Is your title case consistent? (Severity: High)**: `usage`
*Explanation: Could you use the AP Stylebook for title case? Capitalize most words in titles.*
```suggestion
### Basic Usage
```

    steps:
      - uses: actions/checkout@v5
      - name: Run Analysis
        uses: markupai/content-guardian-action@v2
        with:
          markup_ai_api_key: ${{ secrets.MARKUP_AI_API_KEY }}
          github_token: ${{ secrets.GITHUB_TOKEN }}
          target: Marketing Voice
```

> ### Advanced configuration

Author: **Markup AI** detected issues:
- **Consistency / Is your title case consistent? (Severity: High)**: `configuration`
*Explanation: Could you use the AP Stylebook for title case? Capitalize most words in titles.*
```suggestion
### Advanced Configuration
```

    steps:
      - uses: actions/checkout@v5
      - name: Run Analysis
        uses: markupai/content-guardian-action@v2
        with:
          markup_ai_api_key: ${{ secrets.MARKUP_AI_API_KEY }}
          github_token: ${{ secrets.GITHUB_TOKEN }}
          target: ${{ secrets.MARKUP_AI_TARGET_ID }} # accepts ID or display name
          add_commit_status: "true"
          add_review_comments: "true"
          strict_mode: "false"
```

> ## Required tokens

Author: **Markup AI** detected issues:
- **Consistency / Is your title case consistent? (Severity: High)**: `tokens`
*Explanation: Could you use the AP Stylebook for title case? Capitalize most words in titles.*
```suggestion
## Required Tokens
```

> ### API token

Author: **Markup AI** detected issues:
- **Consistency / Is your title case consistent? (Severity: High)**: `token`
*Explanation: Could you use the AP Stylebook for title case? Capitalize most words in titles.*
```suggestion
### API Token
```

- **Required**: Yes
- **Input name**: `markup_ai_api_key`
- **Environment variable**: `MARKUP_AI_API_KEY`
- **Get one**: Sign up at [console.markup.ai](https://console.markup.ai)

> ### GitHub token

Author: **Markup AI** detected issues:
- **Consistency / Is your title case consistent? (Severity: High)**: `token`
*Explanation: Could you use the AP Stylebook for title case? Capitalize most words in titles.*
```suggestion
### GitHub Token
```

| `files-analyzed` | Number of files analyzed                       |
| `results`        | JSON string containing analysis results        |

> ## Event types

Author: **Markup AI** detected issues:
- **Consistency / Is your title case consistent? (Severity: High)**: `types`
*Explanation: Could you use the AP Stylebook for title case? Capitalize most words in titles.*
```suggestion
## Event Types
```

### Push (`on: [push]`)

Analyzes files modified in the push. Updates the commit status with the overall
quality score (numeric mode) or risk label (risk mode).

> ### Pull request (`on: [pull_request]`)

Author: **Markup AI** detected issues:
- **Consistency / Is your title case consistent? (Severity: High)**: `request`
*Explanation: Could you use the AP Stylebook for title case? Capitalize most words in titles.*
```suggestion
### Pull Request (`on: [pull_request]`)
```

> Analyzes files changed in the PR. Posts a single PR comment with the results

Author: **Markup AI** detected issues:
- **Spelling and grammar / Is this the right word in this context? (Severity: High)**: `Analyzes`
*Explanation: Is this the right word in this context? The word or combination of words seems to be wrong in this context. Use a different word or punctuation mark. It could also be a misspelling.*

table and inline review comments for each flagged line.

> ### Manual (`on: [workflow_dispatch]`) and scheduled (`on: [schedule]`)

Author: **Markup AI** detected issues:
- **Consistency / Is your title case consistent? (Severity: High)**: `scheduled`
*Explanation: Could you use the AP Stylebook for title case? Capitalize most words in titles.*
```suggestion
### Manual (`on: [workflow_dispatch]`) and Scheduled (`on: [schedule]`)
```

> Analyzes every supported file in the repository at the current ref. Writes the

Author: **Markup AI** detected issues:
- **Spelling and grammar / Is this word improperly capitalized? (Severity: High)**: `Writes`
*Explanation: Is this word improperly capitalized? It looks like this shouldn’t be capitalized. In English, you usually capitalize the first word of a sentence, proper nouns, and headings only.*
```suggestion
Analyzes every supported file in the repository at the current ref. writes the
```

results to the workflow's job summary.

> ## Risk vs. numeric scoring

Author: **Markup AI** detected issues:
- **Consistency / Is your title case consistent? (Severity: High)**: `numeric`
*Explanation: Could you use the AP Stylebook for title case? Capitalize most words in titles.*
```suggestion
## Risk vs. Numeric scoring
```
- **Consistency / Is your title case consistent? (Severity: High)**: `scoring`
*Explanation: Could you use the AP Stylebook for title case? Capitalize most words in titles.*
```suggestion
## Risk vs. numeric Scoring
```

The action queries `GET /style-agent/config` once per run. If your org has
`style_agent_numeric_scoring` enabled, the action shows:

- per-file quality score (0–100) and per-goal breakdown
- summary average quality across files
- commit status `Quality 78 | Files 4 | Issues 12 (H:1 M:5 L:6)`

> If your org does not have numeric scoring enabled, the action shows:

Author: **Markup AI** detected issues:
- **Tone / Did you use contractions? (Severity: High)**: `does`
*Explanation: Could you use contractions to make it less formal? Use contractions instead of the full form to make your content more conversational.*
```suggestion
If your org doesn’t not have numeric scoring enabled, the action shows:
```

- per-file severity counts (high / medium / low) and a risk label
- an overall risk = highest severity present across files
- commit status `Risk Medium | Files 4 | Issues 12 (H:1 M:5 L:6)`

> In both modes the `outputs.results` JSON contains the full per-file structure so

Author: **Markup AI** detected issues:
- **Clarity / Could you add a comma after the introductory phrase? (Severity: High)**: `In`
*Explanation: Could you add a comma after the introductory phrase? If you use a comma after a prepositional or introductory phrase, your content will be easier to read.*

downstream steps can act on the raw data.

> ## Finding your `target` value

Author: **Markup AI** detected issues:
- **Consistency / Is your title case consistent? (Severity: High)**: `your`
*Explanation: Could you use the AP Stylebook for title case? Capitalize most words in titles.*
```suggestion
## Finding Your `target` value
```
- **Consistency / Is your title case consistent? (Severity: High)**: `value`
*Explanation: Could you use the AP Stylebook for title case? Capitalize most words in titles.*
```suggestion
## Finding your `target` Value
```
